packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -53,6 +55,8 @@ import {
   PickersRangeCalendarHeader,
   PickersRangeCalendarHeaderProps,
 } from '../PickersRangeCalendarHeader';
+import { useIsDateDisabled } from '@mui/x-date-pickers/DateCalendar/useIsDateDisabled';
+import { findFirstEnabledDate, findLastEnabledDate } from '@mui/x-date-pickers/internals/utils/date-utils';
```

Comment:
We and end-users can only use 2nd-level deep imports directly from the package.
Anything deeper does not have a `package.json` file generated and thus will fail.
These could be:
```tsx
// I would argue that this hook could/should be moved to `hooks` folder
import { useIsDateDisabled } from '@mui/x-date-pickers/DateCalendar';
import { findFirstEnabledDate, findLastEnabledDate } from '@mui/x-date-pickers/internals';
```
P.S. You could also just create your own version of `useIsDisabledDate` for this particular case, given how simple that hook is.